packages/config/src/library/v2/common.ts

```diff
@@ -0,0 +1,21 @@
+// eslint-disable-next-line @typescript-eslint/no-empty-interface
+export interface MudConfigInput {
+  /** This interface can be extended by MUD config plugins */
+}
+
+// eslint-disable-next-line @typescript-eslint/no-empty-interface, @typescript-eslint/no-unused-vars
+export interface MudResolvedConfigOutput<configInput extends MudConfigInput = MudConfigInput> {
+  /** This interface can be extended by MUD config plugins */
+}
+
+export interface MudConfigOutput<configInput extends MudConfigInput = MudConfigInput> {
+  resolved: MudResolvedConfigOutput<configInput>;
+}
+
+export type resolveConfig<configInput extends MudConfigInput> = configInput & MudConfigOutput<configInput>;
+
+export const resolveConfig = <configInput extends MudConfigInput>(
```

Comment:
thoughts on `resolveConfig` vs `defineConfig`?

maybe `defineConfig` wraps `resolveConfig` among other things if we e.g. have a separate but built-in codegen config, ECS config, etc?

Reply:
just a leftover i missed, removing